samples/snippets/sample_templates.py

```diff
@@ -0,0 +1,195 @@
+    template.properties.network_interfaces = [network_interface]
+
+    template_client = compute_v1.InstanceTemplatesClient()
+    operation_client = compute_v1.GlobalOperationsClient()
+    op = template_client.insert(project=project_id, instance_template_resource=template)
+    operation_client.wait(project=project_id, operation=op.name)
+
+    return template_client.get(project=project_id, instance_template=template_name)
+
+
+def create_template_from_instance(project_id: str, instance: str, template_name: str) -> compute_v1.InstanceTemplate:
+    """
+    Create a new instance template based on an existing instance.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you use.
+        instance: the instance the new template will be based on. This value uses
+            the following format: "projects/{project}/zones/{zone}/instances/{instance_name}"
+        template_name: name of the new template that is created.
+
+    Returns:
+        InstanceTemplate object describing the new instance template.
+    """
+    disk = compute_v1.DiskInstantiationConfig()
+    # This name must match the name of a disk attached to the instance you are
+    # basing your template on.
+    disk.device_name = "disk-1"
+    # Replace the original disk image used in your instance with a Rocky Linux image.
+    disk.instantiate_from = (
+        compute_v1.DiskInstantiationConfig.InstantiateFrom.CUSTOM_IMAGE
+    )
+    disk.custom_image = "projects/rocky-linux-cloud/global/images/family/rocky-linux-8"
+    # You can override the auto_delete setting.
+    disk.auto_delete = True
+
+    template = compute_v1.InstanceTemplate()
+    template.name = template_name
+    template.source_instance = instance
+    template.source_instance_params = compute_v1.SourceInstanceParams()
+    template.source_instance_params.disk_configs = [disk]
+
+    template_client = compute_v1.InstanceTemplatesClient()
+    operation_client = compute_v1.GlobalOperationsClient()
+    op = template_client.insert(project=project_id, instance_template_resource=template)
+    operation_client.wait(project=project_id, operation=op.name)
+
+    return template_client.get(project=project_id, instance_template=template_name)
+
+
+def create_template_with_subnet(
+    project_id: str, network: str, subnetwork: str, template_name: str
+) -> compute_v1.InstanceTemplate:
+    """
+    Create an instance template that will use a provided subnet.
+
+    Args:
+        project_id: project ID or project number of the Cloud project you use.
+        network: the network to be used in the new template. This value uses
+            the following format: "projects/{project}/global/networks/{network}"
+        subnetwork: the subnetwork to be used in the new template. This value
+            uses the following format: "projects/{project}/regions/{region}/subnetworks/{subnetwork}"
+        template_name: name of the new template that is created.
```

Comment:
nit: "name of the new template that is created." -> "name of the new template to create."

(present tense)